tests/models/test_dashboards.py

```diff
@@ -79,6 +79,6 @@ def create_dashboard():
         create_dashboard()
         create_dashboard()
 
-        results = Dashboard.all(self.factory.org, usr.group_ids, usr.id)
+        results = db.session.scalars(Dashboard.all(self.factory.org, usr.group_ids, usr.id)).all()
 
-        self.assertEqual(2, results.count(), "The incorrect number of dashboards were returned")
+        self.assertEqual(2, len(results), "The incorrect number of dashboards were returned")
```

Comment:
we cannot. principle of db request was changed in 1.4

Reply:
cannot use count in this case. `func.count` is supported only as `select` function argument, it cannot be applied to a `Select` object

Reply:
At first, `func.count` is different from `len`, right? `func.count` will be translated "select count(*) from ..." executed, and return the result. [len](https://docs.python.org/3/library/functions.html#len) returns the length (the number of items) of an object.

Can you update it by following https://docs.sqlalchemy.org/en/20/changelog/migration_20.html ?
```
session.scalar(
  select(func.count()).
  select_from(Dashboard)
)
```

The reason I point this out in detail is because there is a difference in performance between `select count(*) from ...` and len(Array or so on).

`select count(*) from ...` gets the count on the database side and returns it. On the other hand, if you use len(Array or so on) instead, it will return the size of the objects. They are instantiated all even though we would like to get the size only.

In the case where 100 million objects are returned, the performance is quite different. So I would argue that changing SQLAlchemy's count to len is not a good thing.

Reply:
updated, please check

Reply:
Thanks for your update. I confirmed it and it is fine.
However, there are a few other places that have been changed to use `len()` instead of `count()` in this time. Could you please confirm?

Reply:
found it in some tests, just pushed

Reply:
It looks like there are still a few other places that have been changed to use len() instead of count() this time... OK, I will point them out. Please check them.

Reply:
@masayuki038 updated these places as well

Reply:
I finished pointing them out. So, I close this thread.

Reply:
@masayuki038 Thanks for taking the time to do this. :smile: